pyproject.toml

```diff
@@ -16,10 +16,10 @@ python = "^3.6"
 pytest = "~6.0.0"
 jmespath = "^0.10.0"
 portalocker = "^1.7.0"
+pytest-xdist = ">= 1.31.0"
```

Comment:
Can we stick with `^1.31.0`? That will do [>= 1.31.0 but not 2.0](https://python-poetry.org/docs/versions/) so we don't accidentally break things if upstream changes faster than we're ready.

Reply:
we work on xdist 2.0, and thats already out so we don't want to pin to old versions, we want to catch breakage proactively, we should setup a nightly github action to catch any issues against these pins. the lock file already captures repeatable install per se albeit thats source install.

Reply:
I'm not sure what you want me to do here, keep the `>=` or switch to `^`?

Reply:
Fair enough. I'll take up an item to expand our testing on latest deps for our installs